### PR TITLE
Move PluginUpdateManager to core and enable both Toolkit and Q auto-update

### PR DESCRIPTION
**Pull request by andrewyuq**
Create PluginUpdateManager in core module. It will be called on core plugin startup to start auto-updating core, amazonq and toolkit if any of them is detected to be installed. The auto-update setting is shared across all three plugins in the AWS settings pane.

https://github.com/aws/aws-toolkit-jetbrains/assets/89420755/0a1b1253-9778-4e79-8850-d8a3c8f2cc83

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/plugin/PluginUpdateManager.kt [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+import software.aws.toolkits.jetbrains.AwsPlugin
+import software.aws.toolkits.jetbrains.AwsToolkit
+import software.aws.toolkits.jetbrains.settings.PluginSettings
+import software.aws.toolkits.jetbrains.utils.notifyInfo
+import software.aws.toolkits.resources.message
+import software.aws.toolkits.telemetry.Component
+import software.aws.toolkits.telemetry.Result
+import software.aws.toolkits.telemetry.ToolkitTelemetry
+
+abstract class PluginUpdateManager {
```
Comment (rli) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/plugin/PluginUpdateManager.kt:35]: single instance in core that updates all plugins at once

Comment (andrewyuq) [reply plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/plugin/PluginUpdateManager.kt:35]: In this case if user has installed one but not the other, the logic will still need to check the existence of such plugin?

Comment (andrewyuq) [reply plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/plugin/PluginUpdateManager.kt:35]: Also there's a techinical blocker that it needs to open CodeWhisperer setttings or Aws Settings which the core doesn't have access to. We will have to move around files

Comment (manodnyab) [reply plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/plugin/PluginUpdateManager.kt:35]: Should this also update the core plugin whenever we try to update either of the plugins?

Comment (andrewyuq) [reply plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/plugin/PluginUpdateManager.kt:35]: yea, it should also update the core, I will make the change.

Comment (andrewyuq) [reply plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/plugin/PluginUpdateManager.kt]: changes done.

**Comment by rli** [comment]
```
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953   java.lang.NoClassDefFoundError: Could not initialize class software.aws.toolkits.jetbrains.AwsToolkit
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at software.aws.toolkits.jetbrains.services.codewhisperer.CodeWhispererTelemetryServiceTest.setup(CodeWhispererTelemetryServiceTest.kt:93)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at java.base/java.lang.reflect.Method.invoke(Method.java:568)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.internal.runners.statements.RunBefores.invokeMethod(RunBefores.java:33)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at com.intellij.testFramework.ApplicationRule$apply$1.evaluate(FixtureRule.kt:71)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at com.intellij.testFramework.ApplicationRule$apply$1.evaluate(FixtureRule.kt:71)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:42)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:80)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:72)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:119)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:94)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:89)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:62)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at java.base/java.lang.reflect.Method.invoke(Method.java:568)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at jdk.proxy2/jdk.proxy2.$Proxy5.stop(Unknown Source)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.gradle.api.internal.tasks.testing.worker.TestWorker$3.run(TestWorker.java:193)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:113)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:65)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953   Caused by: java.lang.ExceptionInInitializerError: Exception java.nio.file.InvalidPathException: Illegal char <:> [in thread "ApplicationImpl pooled thread 6"]
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at java.base/sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:177)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at java.base/sun.nio.fs.WindowsPath.parse(WindowsPath.java:92)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at java.base/sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:232)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at java.base/java.nio.file.Path.of(Path.java:147)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at java.base/java.nio.file.Paths.get(Paths.java:69)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at software.aws.toolkits.jetbrains.PluginInfo.<init>(AwsToolkit.kt:49)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at software.aws.toolkits.jetbrains.AwsToolkit.<clinit>(AwsToolkit.kt:22)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at software.aws.toolkits.jetbrains.core.plugin.PluginUpdateManager.checkForUpdates(PluginUpdateManager.kt:68)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at software.aws.toolkits.jetbrains.core.plugin.PluginUpdateManager$scheduleAutoUpdate$2$1.run(PluginUpdateManager.kt:54)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:477)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:133)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressSynchronously$9(CoreProgressManager.java:567)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at com.intellij.openapi.progress.impl.ProgressRunner.lambda$new$0(ProgressRunner.java:86)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:250)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:100)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$1(CoreProgressManager.java:221)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at com.intellij.platform.diagnostic.telemetry.helpers.TraceKt.use(trace.kt:46)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:220)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:660)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:735)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:691)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:659)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:79)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:202)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:100)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$5(ProgressRunner.java:250)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at com.intellij.openapi.progress.impl.ProgressRunner$ProgressRunnable.run(ProgressRunner.java:500)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
Windows-Unit-Tests-2024-1 352b9eb1-fc6f-4fb2-9246-23a6ba258953       at java.base/java.lang.Thread.run(Thread.java:840)
```

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/AwsToolkit.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        AwsPlugin.CORE to CORE_PLUGIN_INFO
+    )
+
+    const val GITHUB_URL = "https://github.com/aws/aws-toolkit-jetbrains"
+    const val AWS_DOCS_URL = "https://docs.aws.amazon.com/console/toolkit-for-jetbrains"
+    const val TOOLKIT_PLUGIN_NAME = "AWS Toolkit"
+}
+
+data class PluginInfo(val id: String) {
+    val descriptor: PluginDescriptor? = PluginManagerCore.getPlugin(PluginId.getId(id))
```
Comment (rli) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/AwsToolkit.kt:38]:
```suggestion
    val descriptor: PluginDescriptor?
        get() = PluginManagerCore.getPlugin(PluginId.getId(id))
```

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/AwsToolkit.kt [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import com.intellij.openapi.extensions.PluginDescriptor
+import com.intellij.openapi.extensions.PluginId
+import java.nio.file.Path
+import java.nio.file.Paths
+
+object AwsToolkit {
+    const val TOOLKIT_PLUGIN_ID = "aws.toolkit"
+
+    // TODO: change this to real plugin id
+    const val Q_PLUGIN_ID = "plugin-amazonq"
```
Comment (rli) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/AwsToolkit.kt:17]:
```suggestion
    const val Q_PLUGIN_ID = "amazon.q"
```

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/AwsToolkit.kt [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    // TODO: change this
+    const val CORE_PLUGIN_ID = "plugin-core"
```
Comment (rli) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/AwsToolkit.kt:20]:
```suggestion
    const val CORE_PLUGIN_ID = "aws.toolkit.core"
```

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/AwsToolkit.kt [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+data class PluginInfo(val id: String) {
+    val descriptor: PluginDescriptor? = PluginManagerCore.getPlugin(PluginId.getId(id))
+    val name: String = when (id) {
+        AwsToolkit.TOOLKIT_PLUGIN_ID -> "AWS Toolkit"
+        AwsToolkit.Q_PLUGIN_ID -> "Amazon Q"
+        // TODO: change name
+        AwsToolkit.CORE_PLUGIN_ID -> "AWS Plugin Core"
+        else -> "AWS Toolkit"
```
Comment (rli) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/AwsToolkit.kt:44]: should just make it a constructor param

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/AwsToolkit.kt [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    const val Q_PLUGIN_ID = "plugin-amazonq"
+
+    // TODO: change this
+    const val CORE_PLUGIN_ID = "plugin-core"
+
+    private val TOOLKIT_PLUGIN_INFO = PluginInfo(TOOLKIT_PLUGIN_ID)
+    private val Q_PLUGIN_INFO = PluginInfo(Q_PLUGIN_ID)
+    private val CORE_PLUGIN_INFO = PluginInfo(CORE_PLUGIN_ID)
+
+    val PLUGINS_INFO = mapOf(
```
Comment (rli) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/AwsToolkit.kt:26]: `EnumMap`?

**Comment by andrewyuq** [comment]
https://github.com/aws/aws-toolkit-jetbrains/pull/4258#issuecomment-2057665164
give `path` a lazy get() should resolve this.

**Review by rli — APPROVED** [review]

**Pull request merged**